src/main/inAppNotifications/useInAppNotifications.ts

```diff
@@ -111,5 +113,20 @@ export const useInAppNotifications = () => {
     });
   };
 
-  return { items, isLoading: loading, updateNotificationState };
+  const markNotificationsAsRead = async () => {
+    const ids = items.filter(item => item.state === InAppNotificationState.Unread).map(item => item.id);
+
+    if (ids.length === 0) {
+      return;
+    }
+
+    await markAsRead({
+      variables: {
+        notificationIds: ids,
+      },
+      refetchQueries: [refetchInAppNotificationsQuery()],
```

Comment:
refetching the whole set of notifications might be too heavy for just updating the state.

```suggestion
      update: cache => {
        const existingData = cache.readQuery<InAppNotificationsQuery>({
          query: InAppNotificationsDocument,
        });

        if (existingData) {
          const updatedNotifications = existingData.notifications.map(notification =>
            ids.includes(notification.id)
              ? { ...notification, state: InAppNotificationState.Read }
              : notification
          );

          cache.writeQuery({
            query: InAppNotificationsDocument,
            data: { notifications: updatedNotifications },
          });
        }
      }
```